### PR TITLE
onnx export of per channel fake quantize functions

### LINKED ISSUE #39502: Export fake quantization function to ONNX
## 🚀 Feature

Export fake_quantize_per_tensor_affine and fake_quantize_per_channel_affine functions to ONNX.

## Motivation

To support deploying QAT network by backend like TensorRT  outside pytorch through ONNX, fake quantization needs to be exported to ONNX operator.

## Pitch

fake quantization will be broken into a pair of QuantizeLinear and DequantizeLinear ONNX operator.

## Additional context

Fake quantization is effectively quantize followed dequantize. Have discussed with @raghuramank100 and reached an agreement this is the right way to export. 

Similar functionality has been added to tensorflow-onnx, https://github.com/onnx/tensorflow-onnx/pull/919. 

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

### PR DESCRIPTION
**Pull request by skyw**
Fixes #39502

This PR adds support for exporting **fake_quantize_per_channel_affine** to a pair of QuantizeLinear and DequantizeLinear. Per tensor support was added by PR https://github.com/pytorch/pytorch/pull/39738.

`axis` attribute of QuantizeLinear and DequantizeLinear, which is required for per channel support, is added in opset13 added by https://github.com/onnx/onnx/pull/2772.

[update 1/20/2021]: opset13 is being supported on master, the added function is now properly tested. Code also rebased to new master.

The function is also tested offline with the following code
```python
import torch
from torch import quantization

from torchvision import models
qat_resnet18 = models.resnet18(pretrained=True).eval().cuda()

qat_resnet18.qconfig = quantization.QConfig(
    activation=quantization.default_fake_quant, weight=quantization.default_per_channel_weight_fake_quant)
quantization.prepare_qat(qat_resnet18, inplace=True)
qat_resnet18.apply(quantization.enable_observer)
qat_resnet18.apply(quantization.enable_fake_quant)

dummy_input = torch.randn(16, 3, 224, 224).cuda()
_ = qat_resnet18(dummy_input)
for module in qat_resnet18.modules():
    if isinstance(module, quantization.FakeQuantize):
        module.calculate_qparams()
qat_resnet18.apply(quantization.disable_observer)

qat_resnet18.cuda()

input_names = [ "actual_input_1" ]
output_names = [ "output1" ]


torch.onnx.export(qat_resnet18, dummy_input, "quant_model.onnx", verbose=True, opset_version=13)
```
It can generate the desired graph.

### CODE REVIEW

**Comment by skyw** [comment]
@gottbrath @raghuramank100 @vkuzo @houseroad FYI

File: test/onnx/test_models.py [hunk 1]
```diff
@@ -189,5 +189,25 @@ def test_qat_resnet(self):
 
         self.exportTest(toC(qat_resnet50), toC(x))
 
+    @skipIfUnsupportedMinOpsetVersion(13)
+    def test_qat_resnet_per_channel(self):
+        # Quantize ResNet50 model
+        x = Variable(torch.randn(BATCH_SIZE, 3, 224, 224).fill_(1.0))
```
Comment (vadimkantorov) [inline test/onnx/test_models.py:195]: Is explicit `Variable` still useful?

Comment (skyw) [reply test/onnx/test_models.py:195]: I guess no. I copied from other tests. Will remove it.

**Comment by skyw** [comment]
@houseroad , any more comment? Thanks.

File: test/onnx/test_models.py (hunk 1 above)
Comment (ptrblck) [reply test/onnx/test_models.py:195]: I would suggest to remove the deprecated usage of `Variable`. The other tests could be updated in a separate PR.

Comment (skyw) [reply test/onnx/test_models.py]: Removed `Variable` in all fake quantization ONNX export tests.

**Comment by skyw** [comment]
ONNX runtime now supports opset13 version of QuantizeLinear and DequantizeLinear, https://github.com/microsoft/onnxruntime/pull/4759 on master. Verified Q/DQ with axis generated by this change does is valid ONNX op.

**Comment by skyw** [comment]
TF2ONNX now supports this, https://github.com/onnx/tensorflow-onnx/pull/1081

**Comment by daquexian** [comment]
@houseroad @raghuramank100 Could you please review this PR? It is an important step to enable deploying the PyTorch QAT model on various backends like TensorRT and many mobile frameworks.

**Comment by skyw** [comment]
@spandantiwari , rebased to new master.

**Review by neginraoof — APPROVED** [review]
Thanks for adding this op.

**Review by houseroad — APPROVED** [review]
Looks good.
